loggo/loggo.py

```diff
@@ -352,25 +352,22 @@ def _params_to_dict(self, function, *args, **kwargs):
                 bound.pop('cls')
         return bound
 
-    def _obscure_private_keys(self, dictionary, dict_depth=0):
+    def _obscure_private_keys(self, log_data, dict_depth=0):
         """
         Obscure any private values in a dictionary recursively
         """
-        if not self.private_data:
-            return dictionary
+        too_deep = dict_depth >= self.max_dict_depth
+
+        if not self.private_data or not isinstance(log_data, dict) or too_deep:
```

Comment:
This would probably work just the same with one condition less like this: `if not isinstance(log_data, dict) or too_deep:`